pennylane_lightning/core/src/utils/cuda_utils/cuda_helpers.hpp

```diff
@@ -338,6 +338,17 @@ static std::pair<int, int> getGPUArch(int device_number = 0) {
     return std::make_pair(deviceProp.major, deviceProp.minor);
 }
 
+/**
+ * @brief Get free memory size on GPU device
+ *
+ * @return size_t
+ */
+inline std::size_t getFreeMemorySize() {
```

Comment:
Are we using this somewhere to validate if more space is available?

Reply:
yes, we should specify the maximum workspace size for the `cutensorStatePrepar()` API [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/e606691beb78da859185b618f2a902daaa7718f7/pennylane_lightning/core/src/simulators/lightning_tensor/cutn/CutnBase.hpp#L166-L173)